lib/pbench/server/api/resources/query_apis/metadata_index/namespace_and_rows.py

```diff
@@ -66,29 +52,36 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
         """
         run_id = context["run_id"]
         dataset = context["dataset"]
+        document = self.ES_INTERNAL_INDEX_NAMES.get(json_data["type"])
+        if not document:
+            self.logger.debug("Illegal document name {}", json_data["type"])
+            abort(HTTPStatus.NOT_FOUND, message="Namespace not found")
```

Comment:
This isn't like looking for a user-visible resource, like a Dataset name or a User; this is the client giving an unrecognized keyword. I think this should be `BAD_REQUEST`.

I would have defined your `type` Parameter as a `KEYWORD` with `list(RunIdBase.ES_INTERNAL_INDEX_NAMES.keys())` as the allowed values; which would have already failed with a `BAD_REQUEST` before reaching this code, with an error message consistent with other keyword validation failures.

Reply:
Ah good point